metadata-ingestion/src/datahub/configuration/common.py

```diff
@@ -158,6 +162,26 @@ class IgnorableError(MetaError):
     """An error that can be ignored."""
 
 
+@runtime_checkable
+class ExceptionWithProps(Protocol):
+    def get_telemetry_props(self) -> Dict[str, Any]:
+        ...
+
+
+def should_show_stack_trace(exc: Exception) -> bool:
+    # Unless the exception is a ValidationError or explicitly opts out of stack traces,
+    # we should show the stack trace.
+
+    if isinstance(exc, ValidationError) or isinstance(exc.__cause__, ValidationError):
```

Comment:
This is nice. So any exceptions that we know root cause for does not result in a full exception trace. This makes sense - and in fact is closely aligned with how we should think about source exception handling

Reply:
yup exactly. we're a bit away from enabling this for other exception types, but adding the scaffolding here will help us in the future